cmd/gossamer/configcmd.go

```diff
@@ -60,13 +60,29 @@ var (
 func makeNode(ctx *cli.Context, gen *genesis.GenesisState) (*dot.Dot, *cfg.Config, error) {
 	fig, err := getConfig(ctx)
 	if err != nil {
-		log.Crit("unable to extract required config", "err", err)
 		return nil, nil, err
 	}
 
 	var srvcs []services.Service
 
-	// TODO: trie and runtime
+	if gen == nil {
+		return nil, nil, fmt.Errorf("genesis is nil")
+	}
+
+	if gen.GenesisTrie == nil {
+		return nil, nil, fmt.Errorf("no genesis trie exists")
+	}
+
+	// load runtime code from trie and create runtime executor
+	code, err := gen.GenesisTrie.Get([]byte(":code"))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error retrieving :code from trie: %s", err)
+	}
```

Comment:
I think this should be outside of `makeNode`. This should only cover steps that are required on every startup. It should be assumed at this point that the genesis state is initialized, we just need to load it from disk.